### PR TITLE
refactor(starfish): expose fixed quantiles for heavy latency histograms

### LINKED ISSUE #12334: Compress the heaviest Starfish histogram metrics
Six Starfish histograms account for an estimated 60–80% of the Prometheus time series the node produces. They are histograms whose label sets expand with committee size (~20 buckets × ~70 label values each), so their footprint grows with the network. Reduce their series count without losing the signals the starfish-overview dashboard relies on.

**Key points:**
- Likely dominators (per-peer/per-authority label × bucket expansion): `latency_to_process_stream` (`peer`, 18 buckets), `transaction_synchronizer_fetch_latency` (`peer` × `type`), `proposed_block_ancestors_depth` (`authority`, 20 buckets), `proposed_block_acknowledgments_depth` (`authority`, 20 buckets), `suspended_block_header_time` (`authority`), plus one more to confirm by measurement
- Confirm the exact set and the 60–80% share with a per-metric series-count measurement first
- Dashboard percentile panels use pooled `histogram_quantile` queries; they must keep working after the change

### PR DESCRIPTION
**Pull request by polinikita**
# Description of change

The three heaviest remaining Starfish histograms are only ever read at a few fixed percentiles, yet their per-bucket series scale with committee size. Replace them with a small fixed set of quantiles computed on-node over a sliding window and exposed as `<name>{quantile="..."}` gauges, so the percentile panels keep working without the bucket-series cost.

- Add a `QuantileGauge` / `QuantileGaugeVec` helper: an HDR histogram kept over a ~2-minute sliding window, exposing quantiles `{0.05, 0.33, 0.5, 0.66, 0.95}` as gauge series computed at scrape time (a custom prometheus `Collector`). Empty windows drop their series so panels show gaps, not stale values, and idle label values are removed from the collector entirely.
- Convert `transaction_synchronizer_fetch_latency` (50 buckets × peer × type), `latency_to_process_stream` (18 buckets × peer), and `suspended_block_header_time` (per authority) to quantile gauges. The first two also keep a cheap per-peer mean via `_sum`/`_count`; `suspended_block_header_time` keeps per-authority quantiles.
- The overall fetch-latency quantile gauge merges the `live`/`periodic` split (the two are effectively the same distribution); the by-peer mean keeps the `type` label so the per-peer live/periodic panels stay distinct.
- Update the `starfish-overview` dashboard: percentile panels now read `<name>{quantile="..."}` directly; the per-peer panels use the per-peer mean split by `type`, and the "Bottom 5" send-side panels use the per-peer mean (per-peer/per-authority quantiles are not recoverable once the label is dropped).
- `typed-store` now takes `hdrhistogram` from the workspace, which this PR pins at 7.5.4.

Series per node at committee size 70: these three metrics drop from ~9,900 to ~640; combined with #12368, total Starfish node-metric series go from ~27,700 to ~14,700 (roughly −47%).

## Links to any relevant issues

Part of #12334

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [x] Patch-specific tests (correctness, functionality coverage): unit tests for the sliding-window rotation and idle-window cleanup
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### Release Notes

- [ ] Protocol:
- [x] Nodes (Validators and Full nodes): the `consensus_transaction_synchronizer_fetch_latency`, `consensus_latency_to_process_stream`, and `consensus_suspended_block_header_time` histograms are replaced by per-node gauges exposing fixed quantiles `{0.05, 0.33, 0.5, 0.66, 0.95}` (the first two keep per-peer `_sum`/`_count` means); their `_bucket` series are no longer exposed, so custom `histogram_quantile()` queries over them must switch to the `<name>{quantile="..."}` series.
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] gRPC:

### CODE REVIEW

File: crates/starfish/core/src/metrics.rs [hunk 1]
```diff
@@ -1321,14 +1333,20 @@ impl NodeMetrics {
...
                 "The time taken to fetch transactions from a peer",
-                &["peer", "type"],
-                LATENCY_SEC_BUCKETS.to_vec(),
-                registry;
+                module_path!(),
+                registry,
                 MetricLevel::Warn,
-            ).unwrap(),
+            ),
+            transactions_synchronizer_fetch_latency_by_peer: SumCountVec::register(
```
Comment (bingyanglin) [inline crates/starfish/core/src/metrics.rs:1343]: Dropping `type` made the two fetch-latency panels identical and merges live/periodic into one mean. Maybe keep `type` on the by-peer pair, or delete one panel?

File: crates/starfish/core/src/quantile_gauge.rs [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        for (label_value, window) in windows.iter_mut() {
+            match window.quantiles_seconds() {
+                Some(values) => {
+                    for ((_, quantile), value) in QUANTILES.iter().zip(values) {
+                        self.gauge
+                            .with_label_values(&[label_value.as_str(), quantile])
+                            .set(value);
+                    }
+                }
+                None => {
```
Comment (bingyanglin) [inline crates/starfish/core/src/quantile_gauge.rs]: Idle label windows stay in the map forever. Maybe drop the entry in the `None` branch since it already knows the window is dead?

File: crates/starfish/core/src/quantile_gauge.rs [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    fn new(now: Instant) -> Self {
+        let mut slots = VecDeque::with_capacity(WINDOW_SLOTS);
+        slots.push_back(new_histogram());
+        Self {
+            slots,
+            newest_slot_start: now,
+        }
+    }
+
+    fn rotate(&mut self, now: Instant) {
```
Comment (bingyanglin) [inline crates/starfish/core/src/quantile_gauge.rs:76]: `rotate` has classic off-by-one edge cases and zero tests, but already takes `now` as a param. Maybe add a few unit tests?

**Review by piotrm50 — APPROVED** [review]
That's a huge decrease in metrics!

File: crates/starfish/core/src/metrics.rs (hunk 1 above)
Comment (piotrm50) [reply crates/starfish/core/src/metrics.rs:1343]: that's intended according to PR description:
> Drop the live/periodic split on transaction_synchronizer_fetch_latency — the two are effectively the same distribution.

Comment (polinikita) [reply crates/starfish/core/src/metrics.rs:1343]: Kept the `type` label on the by-peer `_sum`/`_count` pair and restored the live/periodic dashboard panel split in 42b9832591. The overall quantile gauge stays merged across types.

File: crates/starfish/core/src/quantile_gauge.rs (hunk 2 above)
Comment (polinikita) [reply crates/starfish/core/src/quantile_gauge.rs]: Fixed in 42b9832591 — `collect` now retains only live windows, so an idle label value drops both its gauge series and its map entry.

File: crates/starfish/core/src/quantile_gauge.rs (hunk 3 above)
Comment (polinikita) [reply crates/starfish/core/src/quantile_gauge.rs:76]: Added unit tests for `rotate` in 42b9832591: sub-slot no-op, exact slot boundary, multi-slot gap with trimming, whole-window idle reset, and observations aging out of the window.

**Comment by polinikita** [comment]
Checked what else consumes the metrics changed here and in #12368 (all 24 names, searched org-wide and grepped the relevant repos):

Most of them are not affected. There is one internal repo that can be affected, but it is not an important alert.

**Review by bingyanglin — APPROVED** [review]
